services/proxy/pkg/middleware/basic_auth.go

```diff
@@ -1,145 +1,60 @@
 package middleware
 
 import (
-	"fmt"
 	"net/http"
-	"strings"
 
 	"github.com/owncloud/ocis/v2/ocis-pkg/log"
 	"github.com/owncloud/ocis/v2/ocis-pkg/oidc"
 	"github.com/owncloud/ocis/v2/services/proxy/pkg/user/backend"
-	"github.com/owncloud/ocis/v2/services/proxy/pkg/webdav"
 )
 
-// BasicAuth provides a middleware to check if BasicAuth is provided
-func BasicAuth(optionSetters ...Option) func(next http.Handler) http.Handler {
-	options := newOptions(optionSetters...)
-	logger := options.Logger
+type BasicAuthenticator struct {
+	Logger        log.Logger
+	UserProvider  backend.UserBackend
+	UserCS3Claim  string
+	UserOIDCClaim string
+}
 
-	if options.EnableBasicAuth {
-		options.Logger.Warn().Msg("basic auth enabled, use only for testing or development")
+func (m BasicAuthenticator) Authenticate(r *http.Request) (*http.Request, bool) {
+	if isPublicPath(r.URL.Path) {
+		// The authentication of public path requests is handled by another authenticator.
+		// Since we can't guarantee the order of execution of the authenticators, we better
```

Comment:
Why can't we guarantee the order of middlewares? They are executed in the order they are created, or not?

Reply:
Yes, they are executed in the order of the slice that we pass to the Authentication middleware. This exists as a mere protection in case someone changes the order, making it harder for ourselves to shoot us in the foot. 
Removing the check wouldn't really break it but it would result in more unnecessary computation.

Reply:
Ah ok. So we CAN guarantee the order of middlewares we just don't WANT :smile: